dataservice/api/diagnosis/models.py

```diff
@@ -69,65 +68,55 @@ class Diagnosis(db.Model, Base):
                                db.ForeignKey('participant.kf_id'),
                                doc='the participant who was diagnosed',
                                nullable=False)
-    biospecimen_id = db.Column(KfId(),
-                               db.ForeignKey('biospecimen.kf_id'),
-                               doc='the biospecimen which recieved the '
-                               'diagnosis',
-                               nullable=True)
 
 
-def validate_diagnosis_biospecimen(target):
+def validate_diagnosis(target):
     """
     Ensure that both the diagnosis and biospecimen
-    (referred to by biospecimen_id) have the same participant
-
+    have the same participant
     If this is not the case then raise DatabaseValidationError
     """
-    from dataservice.api.biospecimen.models import Biospecimen
     from dataservice.api.errors import DatabaseValidationError
-
-    # Return if diagnosis is None
-    if not target:
-        return
-
-    # Get biospecimen by id
-    bsp = None
-    if target.biospecimen_id:
-        bsp = Biospecimen.query.get(target.biospecimen_id)
-
-    # If biospecimen doesn't exist, return and
+    from dataservice.api.biospecimen.models import Biospecimen
+    # Return if biospecimen is None or
+    # if diagnosis doesn't exist, return and
     # let ORM handle non-existent foreign key
-    if bsp is None:
+    if not target or not target.biospecimens:
         return
-
+    # Get diagnosis by id and bisopecimen by id
+    bsp = Biospecimen.query.get(target.biospecimens[0].kf_id)
```

Comment:
do you need to check all biosecimens?

Reply:
I don't think so. marshmallow seems to be taking care of all objects in the nested request.

Reply:
How does that happen?
Won't `target.biospecimens` be a list of all biospecimens for that diagnosis?